tripal/src/Hook/TripalHooks.php

```diff
@@ -0,0 +1,392 @@
+<?php
+
+namespace Drupal\tripal\Hook;
+
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Hook\Attribute\Hook;
+use Drupal\Core\Render\Markup;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\Core\Url;
+use Drupal\tripal\TripalField\TripalFieldItemBase;
+use Drupal\views\ViewEntityInterface;
+use Drupal\views\ViewExecutable;
+use Drupal\views\ViewsConfigUpdater;
+
+/**
+ * Hook implementations for the Tripal module.
+ */
+class TripalHooks {
+
+  use StringTranslationTrait;
+
+  /**
+   * Implements hook_help().
+   */
+  #[Hook('help')]
+  public function help($route_name, RouteMatchInterface $route_match): string|array|null {
+    switch ($route_name) {
+      // Main module help for the tripal module.
+      case 'help.page.tripal':
+        $output = '<h3>' . $this->t('About') . '</h3>';
+        $output .= '<p>' . $this->t('Tripal is a toolkit to facilitate construction of online genomic, genetic (and other biological) websites.') . '</p>';
+        return ['#markup' => $output];
+    }
+    return NULL;
+  }
+
+  /**
+   * Implements hook_page_attachments().
+   */
+  #[Hook('page_attachments')]
+  public function addPageAttachments(array &$attachments): void {
+    $attachments['#attached']['drupalSettings']['tripal']['vars'] = [
+      'baseurl' => \Drupal::request()->getSchemeAndHttpHost(),
+      'tripal_path' => \Drupal::service('extension.list.module')->getPath('tripal'),
+    ];
+    $attachments['#attached']['library'][] = 'tripal/vars';
+  }
+
+  /**
+   * Implements hook_rebuild().
+   */
+  #[Hook('rebuild')]
+  public function rebuild(): string {
+    \Drupal::service('tripal.rebuild_service')->executeRebuild();
+    // Return value of the module name is only used for phpunit tests.
+    return 'tripal';
+  }
+
+  /**
+   * Implements hook_toolbar().
```

Comment:
Just adding more documentation here to pull some of the tripal lore out of my brain ;-p


```suggestion
   * Implements hook_toolbar().
   *
   * This hook is provided by the Gin Admin theme and we use it to provide
   * the specialized Tripal toolbar in their specialized top menu bar.
```

Reply:
Commit https://github.com/tripal/tripal/pull/2356/commits/8c170d69897e7f771b9b88dac147956d4553284e